src/ledger.rs

```diff
@@ -82,18 +82,64 @@ fn err_bincode_to_io(e: Box<bincode::ErrorKind>) -> io::Error {
     io::Error::new(io::ErrorKind::Other, e.to_string())
 }
 
-fn entry_at<A: Read + Seek>(file: &mut A, at: u64) -> io::Result<Entry> {
+fn raw_entries_at<A: Read + Seek>(
+    file: &mut A,
+    at: u64,
+    num_entries: u64,
+    buf: &mut [u8],
+) -> io::Result<(u64, u64)> {
+    let mut total_read: usize = 0;
+    let mut num_entries_read = 0;
+
     file.seek(SeekFrom::Start(at))?;
 
-    let len = deserialize_from(file.take(SIZEOF_U64)).map_err(err_bincode_to_io)?;
-    trace!("entry_at({}) len: {}", at, len);
+    let len_size = size_of::<u64>();
```

Comment:
I think you'd have less code if you used the index file to add up all the lengths first...

Reply:
yea.. thought of that. It's some extra reads but probably insignificant.

Reply:
a lot less math and deserialization, too

Reply:
Well I wrote it using the index, it didn't come out a lot better, mostly because I couldn't think of an elegant way to handle the last entry which you don't know the length of using the next one because there is no next one.